python/ray/serve/_private/deployment_state.py

```diff
@@ -2805,11 +2818,22 @@ def update(self) -> bool:
         upscales = {}
         downscales = {}
 
+        # NOTE(zcin): If the deployment status is healthy, it should mean
+        # that the number of running replicas at target version is at
+        # the target number. Adding an extra check however to be defensive.
```

Comment:
I would prefer that we don't add this additional check and instead ensure that our states mean exactly what we say they do

Reply:
so this check should only be that all deployments are in `HEALTHY`

Reply:
Discussed offline, added a todo that I will follow up on this next release.